src/content/core/build-service.mdx

```diff
@@ -1,26 +1,138 @@
 
-Once the Okteto CLI is installed, run `okteto context` and select your Okteto instance to download the credentials and certificates required to access the Okteto Build service.
+The Okteto CLI is automatically configured to interact with the Okteto Build service.
+For example, if your have the following Okteto Manifest:
+
+```yaml title="okteto.yaml"
+build:
+  api:
+    context: api
+  frontend:
+    context: frontend
+```
 
-After that, run the `okteto build` command to build and push your images.
+The command `okteto build` will build and automatically push the **api** and **frontend** images to the Okteto Registry:
+
+```bash
+okteto build
+```
+
+```bash
+ i  Using cindy @ okteto.example.com as context
+ i  Building 'api/Dockerfile' in tcp://buildkit.okteto.example.com:443...
+ ...
+ ✓  Image 'registry.okteto.example.com/cindy/api:okteto' successfully pushed
+ i  Building 'frontend/Dockerfile' in tcp://buildkit.okteto.example.com:443...
+ ...
+ ✓  Image 'registry.okteto.example.com/cindy/frontend:okteto' successfully pushed
+```
+
+Read our documentation about the [`build` section](reference/okteto-manifest.mdx#build-object-optional) and the [`okteto build` command](reference/okteto-cli.mdx#build) for more information.
+
+## Smart Builds
+
+Buildkit comes with a local cache to reuse cache layers between image builds. You can configure the [Buildkit cache size and storage class](self-hosted/helm-configuration.mdx#buildkit) when installing Okteto.
+
+But the Buildkit cache has its own limitations. Even if your build is fully cached, you might wait 1 or 2 minutes until your build context is sent to Buildkit, Buildkit pulls your base image, detects that each Dockerfile instruction can be cached, and push the final image again to the Okteto Registry.
+
+Smart Builds is Okteto's feature designed to solve this problem.
+For each image you build, Okteto computes a hash based on the build context, build parameters and build secrets.
+Okteto uses this hash to detect if the image has been already built by another developer. If the image was already built, the build is skipped and the previous image is reused almost immediately.
+
+The way this is implemented is by pushing a tag with the value of the computed hash to the namespace `okteto` in the Okteto Registry.
+For example, if you build the following image in this Okteto Manifest:
+
+```yaml title="okteto.yaml"
+build:
+  api:
+    context: api
+    dockerfile: Dockerfile
+    target: prod
+```
 
+```bash
+okteto build
 ```
-okteto build -t okteto.dev/hello-world:golang .
+
+```bash
+ i  Using cindy @ okteto.example.com as context
+ i  Building 'api/Dockerfile' in tcp://buildkit.okteto.example.com:443...
+ ...
+ => pushing manifest for registry.okteto.example.com/okteto/api:09f8e312c512e566e82
+ ✓  Image 'registry.okteto.example.com/cindy/api:okteto' successfully pushed
+```
+
+A hash is computed based on the content of the `api` folder, the content of the `Dockerfile` file, and the target `prod`.
```

Comment:
Would it make sense to include some build arg in the example to illustrate that? It is mentioned above, but I think adding it to the example might help to understand it

Reply:
Good point, updating!